ng-dev/release/publish/pnpm-versioning.ts

```diff
@@ -0,0 +1,37 @@
+/**
+ * @license
+ * Copyright Google LLC
+ *
+ * Use of this source code is governed by an MIT-style license that can be
+ * found in the LICENSE file at https://angular.io/license
+ */
+
+import {readFile} from 'node:fs/promises';
+import {join} from 'node:path';
+import {existsSync} from 'node:fs';
+
+/**
+ * Class that exposes helpers for fetching and using pnpm
+ * based on a currently-checked out revision.
+ *
+ * This is useful as there is no vendoring/checking-in of specific
+ * pnpm versions, so we need to automatically fetch the proper pnpm
+ * version when executing commands in version branches. Keep in mind that
+ * version branches may have different pnpm version ranges, and the release
+ * tool should automatically be able to satisfy those.
+ */
+export class PnpmVersioning {
+  async isUsingPnpm(repoPath: string) {
+    // If there is only a pnpm lock file at the workspace root, we assume pnpm
+    // is the primary package manager. We can remove such checks in the future.
+    return existsSync(join(repoPath, 'pnpm-lock.yaml')) && !existsSync(join(repoPath, 'yarn.lock'));
+  }
+
+  async getPackageSpec(repoPath: string) {
```

Comment:
`getPackageSpec` as a name was not obviously to me what was being returned. Primarily because I asssociated spec with testing.

I don't know if there is a better naming option, but it was a bit confusing to me on first read.

Reply:
Thanks for doing this!